funcx_endpoint/funcx_endpoint/executors/high_throughput/interchange.py

```diff
@@ -976,11 +976,16 @@ def start(self, poll_period=None):
                             self._ready_manager_queue,
                         )
 
-                        if r["task_id"] in self.task_status_deltas:
-                            del self.task_status_deltas[r["task_id"]]
                         self._ready_manager_queue[manager]["tasks"][task_type].remove(
                             r["task_id"]
                         )
+
+                        # Transfer any outstanding task statuses to the result message
+                        if r["task_id"] in self.task_status_deltas:
+                            r["task_statuses"] += self.task_status_deltas[r["task_id"]]
+                            b_message = dill.dumps(r)
+                            del self.task_status_deltas[r["task_id"]]
+
```

Comment:
I don't follow what's happening.  The intention of this loop appears to be to clean up our references to the task once we know the result has been pushed to the web services.  But now we're storing the task_statuses, reserializing ... and then throwing the serialization away?

That is, I see the deseralization, which makes sense since we need a key (the task id) so that we know _which_ buckets of our data structures to clear:
```python
r = dill.loads(b_message)
```
But now I see a reserialization:
```python
b_message = dill.dumps(r)
```
... and then we iterate and overwrite that serialization?  In which case, why collect the status deltas?

Reply:
Whoops. Got a bit ahead of myself refactoring this and didn't re-test. Fixing it now.